cmd/restic/cmd_snapshots.go

```diff
@@ -61,28 +70,87 @@ func runSnapshots(opts SnapshotOptions, gopts GlobalOptions, args []string) erro
 		}
 	}
 
+	// group by hostname and dirs
```

Comment:
This is duplicate code, I would suggest extracting it into a function which takes a list of snapshots (`[]restic.Snapshot`, called a "slice" in Go) and returns a `map[string][]Snapshot` and an `error`. Let me know if you need help with this.

Reply:
Okay, I extracted it into an own function. Is the style of the new function okay? And shall I remove the duplicate code (in the forget command) in this pull request or shall I open a new one after this got merged?

Reply:
I put the shared code into the internal folder and made the forget command use it too.